ldclient/flag.py

```diff
@@ -14,29 +14,46 @@
 log = logging.getLogger(sys.modules[__name__].__name__)
 
 
-def evaluate(flag, user, store, prereq_events=[]):
+def evaluate(flag, user, store):
+    prereq_events = []
+    if flag.get('on', False):
+        value, prereq_events = _evaluate(flag, user, store)
+        if value is not None:
+            return value, prereq_events
+
+    if 'offVariation' in flag and flag['offVariation']:
+        value = _get_variation(flag, flag['offVariation'])
+        return value, prereq_events
+    return None, prereq_events
+
+
+def _evaluate(flag, user, store, prereq_events=None):
```

Comment:
When we had the default value for prereq_events set to [] it was getting reused across function calls and the array just kept getting appended and we sent back a ton of prereq events.